x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/utils/LicenseUtils.java

```diff
@@ -25,12 +26,18 @@ public static boolean supportedLicense(XPackLicenseState licenseState) {
         return LICENSED_ENT_SEARCH_FEATURE.check(licenseState);
     }
 
-    public static ElasticsearchSecurityException newComplianceException(XPackLicenseState licenseState) {
+    public static ElasticsearchSecurityException newComplianceException(
+        XPackLicenseState licenseState,
+        EnterpriseSearchFeature.Feature feature
+    ) {
         String licenseStatus = licenseState.statusDescription();
 
         ElasticsearchSecurityException e = new ElasticsearchSecurityException(
-            "Current license is non-compliant for search application and behavioral analytics. Current license is {}. "
-                + "Search Applications and behavioral analytics require an active trial, platinum or enterprise license.",
+            "Current license is non-compliant for "
```

Comment:
I wonder if we could reword this a bit, especially for casing concerns? 

Would we be able to do something like `Current license is non-compliant for search applications. Current license is {}. This feature requires an active trial, platinum or enterprise license.`?

I'd want @ioanatia 's input on that though.

Reply:
this message was following the initial product team requirements to not just have a general message that the license is non compliant, but to state the current license level, the required license level and the feature being used.
so I think the wording is fine, the casing should be lower case as it was initially.

Reply:
@ioanatia Do you mean that we can go with the new proposed exception message proposed by @kderusso?

Reply:
Please make sure to get this answered by @ioanatia before merging.

Reply:
current message looks okay with me - still fits the initial requirements from product.
if the feature name is lowercased, this is good to go